### PR TITLE
Failing test case: Valinor behaves differently on ignored additional keys on a single-argument constructor

### PR DESCRIPTION
**Pull request by Ocramius**
In this test, Valinor takes a `__construct(list<int> $values)` and behaves differently:

 * OK with `array{values: list<int>}`
 * OK with `list<int>`
 * BROKEN with `array{values: list<int>, ...}` (additional keys)

This behavior was initially found by @Tigerman55, and affects de-serialization in a complex nested structure.

A workaround that was found to make Valinor consistent is to add an unused argument to the constructor, which seems to make Valinor's behavior more "stable":

```diff
    /** @param list<int> $values */
-    public function __construct(public readonly array $values)
+    public function __construct(public readonly array $values, bool $_ignored = false)
```

### CODE REVIEW

File: tests/Integration/Mapping/Object/ListConstructorMapping.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            ->allowPermissiveTypes()
+            ->allowSuperfluousKeys()
+            ->mapper();
+
+        self::assertSame(
+            [7, 8, 9],
+            $mapper->map(
+                ClassWithSingleListConstructorArgument::class,
+                [
+                    'unrelated_key' => 'this-should-be-ignored-and-have-no-effect',
```
Comment (Ocramius) [inline tests/Integration/Mapping/Object/ListConstructorMapping.php:46]: Notice me senpai

**Comment by Ocramius** [comment]
Something is off.

Locally, I get failures:

```
Valinor on  issue/map-list-constructor-when-unrelated-optional-keys-present via 🐘 v8.2.16 via ❄️  impure (nix-shell) took 2s 
❯ ./vendor/bin/phpunit tests/Integration/Mapping/Object/ListConstructorMapping.php 
PHPUnit 10.5.10 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.16
Configuration: /home/ocramius/Documents/cuyz/Valinor/phpunit.xml.dist

.E.                                                                 3 / 3 (100%)

Time: 00:00.026, Memory: 12.00 MB

There was 1 error:

1) CuyZ\Valinor\Tests\Integration\Mapping\Object\ListConstructorMapping::test_it_will_deserialize_a_hashmap_with_additional_keys_into_an_object_with_a_list_as_constructor_argument
CuyZ\Valinor\Mapper\TypeTreeMapperError: Could not map type `CuyZ\Valinor\Tests\Integration\Mapping\Object\ClassWithSingleListConstructorArgument` with value array{unrelated_key: 'this-should-be-ignored-and-have-no-effect', values: array{…}}. A total of 2 errors were encountered.

/home/ocramius/Documents/cuyz/Valinor/src/Mapper/TypeTreeMapper.php:28
/home/ocramius/Documents/cuyz/Valinor/tests/Integration/Mapping/Object/ListConstructorMapping.php:40

ERRORS!
Tests: 3, Assertions: 6, Errors: 1.
```

In this CI pipeline, all is green. Is the test being run at all?

**Comment by Tigerman55** [comment]
Should the test need to have Test at the end of the filename to be ran?

**Comment by Ocramius** [comment]
> Should the test need to have Test at the end of the filename to be ran?

Thanks: took the occasion to get face and palm to hit each other vigorously :D

**Comment by romm** [comment]
> Thanks: took the occasion to get face and palm to hit each other vigorously :D

Please be gentle, we might still need you here 😁 

Thanks for the report guys, I'll let you review the changes and tell me if that's ok with you before merging. 🙂

**Review by Ocramius — COMMENTED** [review]
Looking good! 👍

**Comment by romm** [comment]
Thanks!

**Pull request merged**